contexts/AutocratContext.tsx

```diff
@@ -40,93 +34,190 @@ export const useAutocrat = () => {
 };
 
 export function AutocratProvider({ children }: { children: ReactNode; }) {
-  const { endpoint } = useNetworkConfiguration();
+  const { network } = useNetworkConfiguration();
   const provider = useProvider();
   const [programVersion, setProgramVersion] = useLocalStorage<ProgramVersion>({
     key: 'program_version',
-    defaultValue: AUTOCRAT_VERSIONS[0],
+    defaultValue: AUTOCRAT_VERSIONS[1],
     serialize: (value) => String(AUTOCRAT_VERSIONS.indexOf(value)),
     deserialize: (value) => AUTOCRAT_VERSIONS[Number(value)],
   });
+
   const { programId, idl } = programVersion!;
-  const dao = useMemo(
-    () =>
-      PublicKey.findProgramAddressSync(
-        [utils.bytes.utf8.encode('WWCACOTMICMIBMHAFTTWYGHMB')],
-        programId,
-      )[0],
-    [programId],
-  );
-  const daoTreasury = useMemo(
-    () => PublicKey.findProgramAddressSync([dao.toBuffer()], programId)[0],
-    [dao, programId],
-  );
-  const autocratProgram = useMemo(
-    () => new Program<AutocratProgram>(idl as AutocratProgram, programId, provider),
-    [provider, programId, idl],
-  );
-  const openbook = useMemo(() => {
-    if (!provider) {
-      return;
-    }
-    return new Program<OpenbookV2>(OPENBOOK_IDL, OPENBOOK_PROGRAM_ID, provider);
-  }, [provider]);
 
-  const { program: openbookTwap } = useOpenbookTwap();
-  const { data: daoStateData } = useQuery({
-    queryKey: ['getDao'],
-    queryFn: () => autocratProgram.account.dao.fetch(dao),
+  // TOOD: Use memo? Or?
+  const autocratProgram = new Program<AutocratProgram>(idl as AutocratProgram, programId, provider);
```

Comment:
you could add a useMemo but the dependency args might look like this:
`[programId.toString()]` you could also add `[programId.toString(), !!provider]`.

useMemo and useEffect dependencies get confusing and are not well documents. But supposedly, if you have like an object, class, function, or array, you  are likely going to get a different reference on each render potentially, and react built-in hooks don't check everything out about your object/function or array to see if it's the same exact values. It just does this `depPrevious === depCurrent` so if it's two different objects they might have the same values but the variables are referencing different objects so it will still return false and re-run the hook. That's why with complex objects I just add `!!object` which converts it to boolean because I know the hook won't run too much and will run enough.

Sorry for the text wall